cypress/integration/page.spec.js

```diff
@@ -1,7 +1,132 @@
-// WRITE TESTS HERE
+/// <reference types="cypress" />
 
 describe('Page', () => {
-  it('should be visitable', () => {
-    cy.visit('/');
+  const apiUrl = 'https://mate.academy/students-api/'
+
+  beforeEach(() => {
+    cy.visit('/')
+  });
+
+  it('user details should be updated after selecting a different user', () => {
+    cy.intercept(`${apiUrl}` + 'users/5', { fixture: 'user5' });
+    cy.intercept(`${apiUrl}` + 'users/6', { fixture: 'user6' });
+
+    cy.get('button')
+      .contains('5')
+      .click();
+
+    cy.getByDataCy('userName')
+      .should('have.text', 'Chelsey Dietrich');
+
+    cy.get('button')
+      .contains('6')
+      .click();
+
+    cy.getByDataCy('userName')
+      .should('have.text', 'Mrs. Dennis Schulist');
+  });
+
+  it('no request to the server after selecting the same user again', () => {
+    cy.intercept(`${apiUrl}` + 'users/5', cy.spy().as('apiCall'));
+
+    cy.get('button')
+      .contains('5')
+      .click()
+      .click();
+
+    cy.get('@apiCall')
+      .its('callCount')
+      .should('equal', 1);
+  });
+
+  it('selected user is cleared after clicking "Clear" button', () => {
+    cy.intercept(`${apiUrl}` + 'users/5', { fixture: 'user5' });
+
+    cy.get('button')
+      .contains('5')
+      .click();
+
+    cy.getByDataCy('userName')
+      .should('have.text', 'Chelsey Dietrich');
+
+    cy.get('button').contains(/[A-z]lear/)
+      .click();
+
+    cy.getByDataCy('userName')
+      .should('not.exist');
+  });
+
+  it('todos can be filtered by title', () => {
+    cy.fixture('todos.json').as('todosList');
+
+    cy.get('@todosList')
```

Comment:
no, you should mock it in the same way as users.
Also, in my mind, it is better to mock all 3 requests in "before each" hook.

You don't need to send a request to receive this data in the test. Just import the file if you don't want to duplicate values from it.(Avoid repeating is a good idea)

Reply:
added mocking to the before each hook